File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        // 1, 1, 2, 8
        // 48, 24, 6, 1
        int[] t = new int[nums.length];
		t[0] = 1;
		
		int[] t1 = new int[nums.length+1];
		t1[nums.length] = 1;
		
		
		for(int i = 1; i < nums.length; i++){
			t[i] = nums[i-1] * t[i-1];
			
        }
        for(int i = nums.length-1; i >= 0; i--) {
        	t1[i] = nums[i] * t1[i+1];
        }
		
		
		for(int i = 0; i < nums.length-1; i++) {
			t[i] = t[i] * t1[i+1];
		}
        return t;
    }
}  

File: Data Structures & Algorithms/subsets/submission-0.cpp
class Solution {
   public:
    void subset(vector<int>& a, vector<int>& ans, int i, vector<vector<int>> &k) {
        if (i == a.size()) {
            k.push_back(ans);
            return;
        }

        ans.push_back(a[i]);
        subset(a, ans, i + 1, k);

        ans.pop_back();
        subset(a, ans, i + 1, k);
    }

    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> vec; 
        vector<int> ans;
        subset(nums, ans, 0, vec);
        return vec;
    }
};
